### ISSUE TITLE
Possible logic error when pausing

### ISSUE THREAD

**Issue by wwarby** [rgergely/steps2fit#1]
It's quite possible that I'm mistaken here, but in reading your codebase it seems as though you have a logic bug that would cause you to record steps that occurred whilst the activity is paused.

As I see it, the intention is that if I were to walk 500 steps with the timer running, then pause the activity and walk another 500 steps, then unpause the activity and walk a final 500 steps, I would expect to see 1000 steps in total recorded in my activity. But my hunch is that in your data field what would happen is the steps would be ticking up to 500 nicely, then stop at 500 when I paused, but when I resume a bit later, the step count would immediately leap up to 1000, and climb to 1500 by the end of my walk.

I haven't downloaded the data field to try it, but suffice to say that either my implementation of the same feature is way over-complicated, or you have a bug (or perhaps you intended that steps that occurred whilst paused should be counted after unpause - but that seems unlikely). Anyway, would be interested to know what you think. My repo is at https://github.com/wwarby/walker if you're interested to see how I dealt with the same issue and although I've tested for that exact scenario, my data field is hot off the press and I wouldn't swear that it's bug free at the moment :)

**Comment by eversonl** [comment]
You are correct when the walk is paused to resume later the steps between (those which happen while the activity is paused) are added back. 

However, I believe this is only the case when activity is paused to 'resume later'. Not when paused while the activity is 'active'
